### PR TITLE
bugfix/22052-drilldown-breadcrumb-label

### LINKED ISSUE #22052: Drilldown breadcrumb showing wrong label when scrollbar: enabled
#### Expected behaviour
After scrolling to the right then clicking on any item to drill down. it should show the proper label of the selected parent item.

#### Actual behaviour
After scrolling to the right then clicking on any item to drill down. it shows random label of the parent series.

#### Live demo with steps to reproduce
https://jsfiddle.net/ut3saxqy/2/

#### Product version
Highstock latest version

#### Affected browser(s)
All

### PR DESCRIPTION
**Pull request by Xillez**
Fixed #22052, drilldown breadcrumb showed the wrong label when scrollbar was enabled

~~Fixed #22052, Drilling to wrong point when scrolled with scrollbar~~

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208629386042303

### CODE REVIEW

File: ts/Extensions/Drilldown/Drilldown.ts [hunk 1]
```diff
@@ -396,7 +396,7 @@ class ChartAdditions {
                 { colorIndex: pick(point.colorIndex, oldSeries.colorIndex) } :
                 { color: point.color || oldSeries.color },
             levelNumber = oldSeries.options._levelNumber || 0,
-            pointIndex = oldSeries.points.indexOf(point);
+            pointIndex = oldSeries.data.indexOf(point);
```
Comment (bre1470) [inline ts/Extensions/Drilldown/Drilldown.ts:399]: I do not think that is correct, should fix https://github.com/highcharts/highcharts/blob/158e0c0c2694a7e24fb0f49ec03458b2b5b1690e/ts/Extensions/Drilldown/Drilldown.ts#L460 instead to keep support for ranges.

**Review by bre1470 — CHANGES_REQUESTED** [review]

File: ts/Extensions/Drilldown/Drilldown.ts (hunk 1 above)
Comment (Xillez) [reply ts/Extensions/Drilldown/Drilldown.ts:399]: @bre1470 Would using point.options & point.index work? or are those in reference to something else than the point array?

Comment (bre1470) [reply ts/Extensions/Drilldown/Drilldown.ts:399]: `Series.data` includes all series points while `Series.points` includes only series points that are supposed to be rendered for the current view / render loop.

**Review by bre1470 — APPROVED** [review]
Looks great! 👍

**Review by hubertkozik — APPROVED** [review]

**Review by KacperMadej — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**